Makefile

```
@@ -349,8 +352,36 @@ $(PROTOC):
 	unzip -oq ${PROTOC_ZIP} -d $(CACHE) 'include/*'; \
 	rm -f ${PROTOC_ZIP})
 
+grpc-gateway:
+	@echo "Installing protoc-gen-grpc-gateway..."
+	@rm -f $@
+	@curl -o "${GOBIN}/protoc-gen-grpc-gateway" -L \
+	"https://github.com/grpc-ecosystem/grpc-gateway/releases/download/v${PROTOC_GRPC_GATEWAY_VERSION}/${PROTOC_GRPC_GATEWAY_BIN}"
+	chmod +x "${GOBIN}/protoc-gen-grpc-gateway"
+
+protoc-swagger:
+ifeq (, $(shell which protoc-gen-swagger))
+	@echo "Installing protoc-gen-swagger..."
+	@go install github.com/grpc-ecosystem/grpc-gateway/protoc-gen-swagger
+	@npm install -g swagger-combine
+else
+	@echo "protoc-gen-swagger already installed; skipping..."
+endif
+
 .PHONY: proto-tools
-proto-tools: cache-setup $(BUF) $(PROTOC)
+proto-tools: cache-setup $(BUF) $(PROTOC) grpc-gateway protoc-swagger
 
 tools-clean:
 	rm -rf $(CACHE)
+
+proto-swagger-gen:
+	./script/protoc-swagger-gen.sh
+
+update-swagger-docs:
+	statik -src=client/grpc-gateway -dest=client/grpc-gateway -f -m
```

lets use `client/docs` for docs.
```suggestion
	statik -src=client/ -dest=client/docs -f -m
```
I am still working on a PR upstream to fix swagger API registration.